src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1034,14 +989,12 @@ int Terminal::ViewEndIndex() const noexcept
 // _VisibleStartIndex is the first visible line of the buffer
 int Terminal::_VisibleStartIndex() const noexcept
 {
-    return _inAltBuffer() ? ViewStartIndex() :
-                            std::max(0, ViewStartIndex() - _scrollOffset);
+    return _inAltBuffer() ? 0 : std::max(0, _mutableViewport.Top() - _scrollOffset);
 }
 
 int Terminal::_VisibleEndIndex() const noexcept
 {
-    return _inAltBuffer() ? ViewEndIndex() :
-                            std::max(0, ViewEndIndex() - _scrollOffset);
+    return _inAltBuffer() ? _altBufferSize.height - 1 : std::max(0, _mutableViewport.BottomInclusive() - _scrollOffset);

```

Comment:
I'm not sure if I should revert these changes... I felt like the extra `_inAltBuffer()` check in the nested calls is redundant and inlining the code would make it clearer, but realistically this doesn't make any difference, right? Idk...

Reply:
jeez, yeah, i don't get why these were the way they are. go for it.